modifier/infill.go

```diff
@@ -25,182 +25,168 @@ func NewInfillModifier(options *data.Options) handler.LayerModifier {
-				} else if i > layerNr {
-					// if we are below layer 0 stop calculation
-					break
-				} else {
-					// else calculate the difference and use it
-					parts, err = partDifference(insetPart, layers[layerNr-1-i])
-					if err != nil {
-						return err
+					// union the parts if needed
+					if len(bottomInfillParts) == 0 {
+						bottomInfillParts = parts
+					} else {
+						var ok bool
+						bottomInfillParts, ok = c.Union(bottomInfillParts, parts)
+						if !ok {
+							return errors.New("could not union bottom parts")
+						}
 					}
 				}
 
-				// union the parts if needed
-				if len(bottomInfillParts) == 0 {
-					bottomInfillParts = parts
-				} else {
-					var ok bool
-					bottomInfillParts, ok = c.Union(bottomInfillParts, parts)
-					if !ok {
-						return errors.New("could not union bottom parts")
+				// calculate the difference with the layers above
+				for i := 0; i < m.options.Print.NumberTopLayers; i++ {
+					var parts []data.LayerPart
+					if layerNr+i == len(layers)-1 {
+						// if it's the last layer, use the whole layer
+						parts = []data.LayerPart{insetPart}
+					} else if layerNr+1+i >= len(layers) {
+						// if we are above the top layer stop calculation
+						break
+					} else {
+						// else calculate the difference and use it
+						parts, err = partDifference(insetPart, layers[layerNr+1+i])
+						if err != nil {
+							return err
+						}
 					}
-				}
-			}
 
-			// calculate the difference with the layers above
-			for i := 0; i < m.options.Print.NumberTopLayers; i++ {
-				var parts []data.LayerPart
-				if layerNr+i == len(layers)-1 {
-					// if it's the last layer, use the whole layer
-					parts = []data.LayerPart{insetPart}
-				} else if layerNr+1+i >= len(layers) {
-					// if we are above the top layer stop calculation
-					break
-				} else {
-					// else calculate the difference and use it
-					parts, err = partDifference(insetPart, layers[layerNr+1+i])
-					if err != nil {
-						return err
+					// union the parts if needed
+					if len(topInfillParts) == 0 {
+						topInfillParts = parts
+					} else {
+						var ok bool
+						topInfillParts, ok = c.Union(topInfillParts, parts)
+						if !ok {
+							return errors.New("could not union top parts")
+						}
 					}
 				}
 
-				// union the parts if needed
-				if len(topInfillParts) == 0 {
-					topInfillParts = parts
-				} else {
-					var ok bool
-					topInfillParts, ok = c.Union(topInfillParts, parts)
-					if !ok {
-						return errors.New("could not union top parts")
+				// 2. Exset the area which needs infill to generate the internal overlap of top and bottom layer.
+				var internalOverlappingBottomParts, internalOverlappingTopParts []data.LayerPart
+				for _, bottomPart := range bottomInfillParts {
+					overlappingParts, err := calculateOverlapPerimeter(bottomPart, m.options.Print.InfillOverlapPercent+m.options.Print.AdditionalInternalInfillOverlapPercent, m.options.Printer.ExtrusionWidth)
```

Comment:
I'd create an own variable for `m.options.Print.InfillOverlapPercent + m.options.Print.AdditionalInternalInfillOverlapPercent`.